lib/components/user/phone-change-form.tsx

```diff
@@ -0,0 +1,182 @@
+// - Typing a phone number that doesn't match the configured phoneNumberRegEx
+//   => invalid (red) alert.
+const phoneValidationSchema = yup.object({
+  phoneNumber: yup
+    .string()
+    .required()
+    .test(
+      'phone-number-format',
+      'invalidPhoneNumber', // not directly shown.
+      (value?: string) => value && isPossiblePhoneNumber(value)
+    )
+})
+
+interface Fields {
+  phoneNumber: string
+}
+
+export type PhoneChangeSubmitHandler = (
+  values: Fields | MouseEvent<Button>
+) => void
+
+interface Props {
+  isSubmitting: boolean
+  onCancel: () => void
+  onSubmit: PhoneChangeSubmitHandler
+  phoneFormatOptions: {
+    countryCode: string
+  }
+  showCancel?: boolean
+}
+
+type InnerProps = FormikProps<Fields> & Props
+
+const InnerPhoneChangeForm = ({
+  errors, // Formik
+  handleBlur, // Formik
+  handleChange, // Formik
+  isSubmitting,
+  onCancel,
+  phoneFormatOptions,
+  showCancel,
+  touched, // Formik
+  values // Formik
+}: InnerProps) => {
+  const intl = useIntl()
+  const formId = 'phone-change-form'
+  const showPhoneError = errors.phoneNumber && touched.phoneNumber
+
+  const handleEscapeKey = useCallback(
+    (e: KeyboardEvent<FormGroup>) => {
+      if (e.key === 'Escape' && showCancel && typeof onCancel === 'function') {
+        e.preventDefault()
+        // Cancel editing when user presses ESC from the phone number field.
+        onCancel()
+      }
+    },
+    [onCancel, showCancel]
+  )
+
+  const handlePhoneChange = useCallback(
+    (newNumber) =>
+      handleChange({
+        target: {
+          name: 'phoneNumber',
+          value: newNumber
+        }
+      }),
+    [handleChange]
+  )
+
+  return (
+    <FormGroup
+      // Handle ESC key from anywhere in this element.
+      onKeyDown={handleEscapeKey}
+      validationState={showPhoneError ? 'error' : null}
+    >
+      {/* Set up an empty Formik Form without inputs, and link inputs using the form id.
+          (A submit button within will incorrectly submit the entire page instead of just the subform.)
+          The containing Formik element will watch submission of the form. */}
+      <Form id={formId} noValidate />
+      <ControlLabel htmlFor="phone-number">
+        <FormattedMessage id="components.PhoneNumberEditor.prompt" />
+      </ControlLabel>
+      <ControlStrip>
+        <InlinePhoneInput
+          aria-invalid={showPhoneError}
+          aria-required
+          className="form-control"
+          country={phoneFormatOptions.countryCode}
+          form={formId}
+          id="phone-number"
+          name="phoneNumber"
+          onBlur={handleBlur}
+          onChange={handlePhoneChange}
+          placeholder={intl.formatMessage({
+            id: 'components.PhoneNumberEditor.placeholder'
+          })}
+          type="tel"
+          value={values.phoneNumber}
+        />
+
+        <Button
+          bsStyle="primary"
+          disabled={isSubmitting}
+          form={formId}
+          type="submit"
```

Comment:
This is a good one. Per that attribute's documentation, `aria-busy` should be applied to a `live` element, in this case that would be the element with `role='status'`. It looks like the best use of `aria-busy` would be on an element with `role='alert'`, where the alert text would be like "New phone number submitted for verification." This would replace the current status element. What do you think @AdrianaCeric?

Reply:
Good catch. You could put an `aria-busy="true"` on line 143. Then, within another InvisibleAl11y tag, you can put an `aria-busy="false"` a long with the `role="alert"` you mentioned describing the phone number is submitted plus/minus variations of that.

Reply:
I'll go for it, although I don't think we need to apply it to the button.

Reply:
Updated in https://github.com/opentripplanner/otp-react-redux/pull/799/commits/703679d7751a2758fa008e57ce50b993c6fba5e3 and https://github.com/opentripplanner/otp-react-redux/pull/799/commits/94ef859eb59938cafac8650fa5684665ff7e0024. Any thoughts @AdrianaCeric?